src/main/java/com/nutomic/syncthingandroid/activities/FolderPickerActivity.java

```diff
@@ -74,6 +78,17 @@ public static Intent createIntent(Context context, String currentPath) {
         return intent;
     }
 
+    //Creates intent with a specified root directory that is used to limit what directories the user can browse.
+    public static Intent createIntentWithRootDir(Context context, String currentPath, String rootDir) {
+        Intent intent = createIntent(context, currentPath);
+
+        if(!TextUtils.isEmpty(rootDir)) {
+            intent.putExtra(EXTRA_ROOT_DIRECTORY, rootDir);
+        }
```

Comment:
I wanted to avoid an empty string "", but I guess that wouldn't actually be a problem.